src/main/java/io/jenkins/plugins/coverage/CoveragePullRequestMonitoringPortlet.java

```diff
@@ -0,0 +1,122 @@
+package io.jenkins.plugins.coverage;
+
+import hudson.Extension;
+import hudson.model.Run;
+import io.jenkins.plugins.coverage.targets.CoverageElement;
+import io.jenkins.plugins.coverage.targets.CoverageResult;
+import io.jenkins.plugins.coverage.targets.Ratio;
+import io.jenkins.plugins.monitoring.MonitorPortlet;
+import io.jenkins.plugins.monitoring.MonitorPortletFactory;
+import org.kohsuke.stapler.bind.JavaScriptMethod;
+
+import java.util.*;
+
+/**
+ * A portlet that can be used for the pull-request-monitoring dashboard
+ * (https://github.com/jenkinsci/pull-request-monitoring-plugin).
+ *
+ * It renders the aggregated line, method and instruction coverage in a stacked bar chart.
+ *
+ * @author Simon Symhoven
+ */
+public class CoveragePullRequestMonitoringPortlet implements MonitorPortlet {
+    private final CoverageAction action;
+
+    /**
+     * Creates a new {@link CoveragePullRequestMonitoringPortlet}.
+     *
+     * @param action
+     *          the {@link CoverageAction} of corresponding run.
+     */
+    public CoveragePullRequestMonitoringPortlet(final CoverageAction action) {
+        this.action = action;
+    }
+
+    @Override
+    public String getTitle() {
+        return action.getDisplayName();
+    }
+
+    @Override
+    public String getId() {
+        return "code-coverage";
+    }
+
+    @Override
+    public int getPreferredWidth() {
+        return 600;
+    }
+
+    @Override
+    public int getPreferredHeight() {
+        return 400;
+    }
+
+    @Override
+    public String getIconUrl() {
+        return null;
+    }
+
+    @Override
+    public Optional<String> getDetailViewUrl() {
+        return Optional.ofNullable(action.getUrlName());
+    }
+
+    /**
+     * Transform the result map of {@link CoverageResult} to a list of {@link CoverageResult.JSCoverageResult}.
+     *
+     * @return
+     *          the transformed list.
+     */
+    @JavaScriptMethod
+    public List<CoverageResult.JSCoverageResult> getResults() {
+        List<CoverageResult.JSCoverageResult> results = new LinkedList<>();
+
+        for (Map.Entry<CoverageElement, Ratio> c : action.getResult().getResults().entrySet()) {
+            String name = c.getKey().getName();
+            if ("Conditional".equals(name) || "Line".equals(name) || "Instruction".equals(name)) {
```

Comment:
It seems a little bit fragile to compare by string. I think `LINE` and `CONDITIONAL` are constants and can be reused. 

And I rarely use instruction since normally one is looking at the source code line by line. So for me it would be sufficient to use only line and branch (which is called conditional here).